File: C# 2/Arrays/MaximalSequence/MaximalSequence.cs
//  Problem 4. Maximal sequence
//  Write a program that finds the maximal sequence of equal elements in an array.

using System;

class MaximalSequence
{
    static void Main()
    {
        Console.WriteLine("enter the length of the array");
        int length = int.Parse(Console.ReadLine());
        int[] arr = new int[length];
        for (int i = 0; i < length; i++)
        {
            arr[i] = int.Parse(Console.ReadLine());
        }

        int bestLength = 0;
        int bestElement = 0;
        int currentLength = 0;
        int currentElement = arr[0];

        for (int i = 0; i < length; i++)
        {
            if(currentElement==arr[i])
            {
                currentLength++;
            }
            else
            {
                currentElement = arr[i];
                currentLength = 1;
            }

            if(currentLength>bestLength)
            {
                bestLength = currentLength;
                bestElement = currentElement;
            }
        }
        Console.WriteLine();
        Console.WriteLine("maximal sequence");
            for(int i=0;i<bestLength;i++)
            {
                Console.Write(bestElement + " ");
            }



    }
}

File: JS/Arrays/MaximalSequence.js
// Write a script that finds the maximal sequence of equal elements in an array.

var arr = [2, 1, 1, 2, 3, 3, 2, 2, 2,1, 2],
    i = 0,
    j= 0,
    count = 0,
    maxCount = 0,
    digit = 0;

for(i; i < arr.length-1; i +=1) {
    if(arr[i]===arr[i+1]) {
        count+=1;
        if(count>maxCount) {
            maxCount=count;
            digit=arr[i];
        }

    } else {
        count = 0;
    }
}

for(j ; j < maxCount+1;j+=1) {
    console.log(digit);
}